### PR TITLE
Add timestamp to IConnected and IClient

### LINKED ISSUE #7152: Add timestamp to IConnected message sent on "connect_document_success"
For GC sweep phase, we need the current time in the client for two purposes:
1. Ensuring that a client session can only last for a specific time. After that, the session becomes read-only.
2. Calculating how long objects have been unreferenced for.
For this, we need the initial time when a client was loaded. We can then use this time as reference to get the current local time. This will be the timestamp to the `IConnected` message.

### PR DESCRIPTION
**Pull request by GaryWilber**
Add `timestamp` property to `IConnected` and `IClient` objects. The timestamp is the time the client connected.

Fixes #7152

### CODE REVIEW

**Review by tanviraumi — APPROVED** [review]

File: server/routerlicious/packages/lambdas/src/alfred/index.ts [hunk 1]
```diff
@@ -236,12 +236,17 @@ export function configureWebSocketServices(
                 return handleServerError(logger, errMsg, claims.documentId, claims.tenantId);
             }
 
+            const connectedTimestamp = Date.now();
```
Comment (agarwal-navin) [inline server/routerlicious/packages/lambdas/src/alfred/index.ts:239]: Are there any mechanisms running on the server to ensure this time is in sync with the timestamp that is added to ops? These two services may run on different machines right.

Comment (GaryWilber) [reply server/routerlicious/packages/lambdas/src/alfred/index.ts:239]: The timestamp for ops is handled by alfred, the same one creating that connectedTimestmap. So they will be in sync

Comment (agarwal-navin) [reply server/routerlicious/packages/lambdas/src/alfred/index.ts:239]: Oh okay, great. I thought deli adds it for ops.

Comment (vladsud) [reply server/routerlicious/packages/lambdas/src/alfred/index.ts:239]: But I believe that does not add any additional guarantees, as two front-ends may have time drift, and as such two clients connecting to document may assume different order RE who joined first if they use timestamps vs. order they see clients added to audience / quorum.
Maybe other way to say it - time may go backwards when looking at op stream.
Is my understanding correct here?

Comment (agarwal-navin) [reply server/routerlicious/packages/lambdas/src/alfred/index.ts:239]: Basically, does this mean different clients will see different timestamp for the same op? Which timestamp does the server use for the messages it stores?

Comment (GaryWilber) [reply server/routerlicious/packages/lambdas/src/alfred/index.ts:239]: The frontends decide on the timestamp for the submitted ops. When deli sequences ops, it uses the timestamp sent from the frontend for the op timestamp. That is the only timestamp for ops - all clients will see that same op.
For vlads question - yes time may go backwards in the op stream if different frontends are submitting ops and are out of sync

**Pull request merged**